client/src/components/History/Multiple/MultipleViewList.vue

```diff
@@ -1,3 +1,62 @@
+<script setup>
+import { computed, ref } from "vue";
+import VirtualList from "vue-virtual-scroll-list";
+import MultipleViewItem from "./MultipleViewItem";
+import { useHistoryStore } from "stores/historyStore";
+import SelectorModal from "components/History/Modals/SelectorModal";
+import { useAnimationFrameScroll } from "composables/sensors/animationFrameScroll";
+import { useAnimationFrameResizeObserver } from "composables/sensors/animationFrameResizeObserver";
+
+const historyStore = useHistoryStore();
+
+const props = defineProps({
+    histories: {
+        type: Array,
+        required: true,
+    },
+    currentHistory: {
+        type: Object,
+        required: true,
+    },
+    handlers: {
+        type: Object,
+        required: true,
+    },
+    filter: {
+        type: String,
+        default: "",
+    },
+});
+
+const isScrollable = ref(false);
+const scrollContainer = ref(null);
+const { arrived } = useAnimationFrameScroll(scrollContainer);
+useAnimationFrameResizeObserver(scrollContainer, ({ clientSize, scrollSize }) => {
+    isScrollable.value = scrollSize.width >= clientSize.width + 1;
+});
+const scrolledLeft = computed(() => !isScrollable.value || arrived.left);
+const scrolledRight = computed(() => !isScrollable.value || arrived.right);
+
+const selectedHistories = computed(() => historyStore.pinnedHistories);
+
+function removeHistoryFromList(history) {
+    historyStore.unpinHistory(history.id);
+}
+
+if (!selectedHistories.value.length) {
+    historyStore.pinHistory(props.histories[0].id);
+}
+
+function addHistoriesToList(histories) {
+    histories.forEach((history) => {
+        const historyExists = selectedHistories.value.find((h) => h.id === history.id);
+        if (!historyExists) {
+            historyStore.pinHistory(history.id);
+        }
+    });
+}
+</script>
```

Comment:
`MultipleViewList` can also be converted to typescript, by ignoring the missing typedefs

```suggestion
<script setup lang="ts">
import { computed, ref, type Ref } from "vue";
//@ts-ignore missing typedefs
import VirtualList from "vue-virtual-scroll-list";
import MultipleViewItem from "./MultipleViewItem.vue";
import { useHistoryStore } from "@/stores/historyStore";
import SelectorModal from "@/components/History/Modals/SelectorModal.vue";
import { useAnimationFrameScroll } from "@/composables/sensors/animationFrameScroll";
import { useAnimationFrameResizeObserver } from "@/composables/sensors/animationFrameResizeObserver";

const historyStore = useHistoryStore();

interface History {
    id: string;
}

const props = withDefaults(
    defineProps<{
        histories: History[];
        currentHistory: History;
        // todo: stricter typedef for handlers, when MultipleViewItem is refactored
        handlers: { [handler: string]: Function };
        filter?: string;
    }>(),
    {
        filter: "",
    }
);

const scrollContainer: Ref<HTMLElement | null> = ref(null);
const { arrived } = useAnimationFrameScroll(scrollContainer);

const isScrollable = ref(false);
useAnimationFrameResizeObserver(scrollContainer, ({ clientSize, scrollSize }) => {
    isScrollable.value = scrollSize.width >= clientSize.width + 1;
});

const scrolledLeft = computed(() => !isScrollable.value || arrived.left);
const scrolledRight = computed(() => !isScrollable.value || arrived.right);

const selectedHistories: Ref<History[]> = computed(() => historyStore.pinnedHistories);

function removeHistoryFromList(history: History) {
    historyStore.unpinHistory(history.id);
}

if (!selectedHistories.value.length) {
    historyStore.pinHistory(props.histories[0].id);
}

function addHistoriesToList(histories: History[]) {
    histories.forEach((history) => {
        const historyExists = selectedHistories.value.find((h) => h.id === history.id);
        if (!historyExists) {
            historyStore.pinHistory(history.id);
        }
    });
}
</script>
```